src/main/java/com/cognifide/controller/BookController.java

```diff
@@ -0,0 +1,63 @@
+package com.cognifide.controller;
+
+import com.cognifide.domain.Book;
+import com.cognifide.repository.BookRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by cognifide.nowak on 04.04.2016.
+ */
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+    public Book get(@PathVariable("id") int id) {
+        return bookRepository.findOne(id);
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public List<Book> listAll() {
+        return bookRepository.listAll();
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
+    public Book delete(@PathVariable("id") int id) {
+        return bookRepository.delete(id);
+    }
+
+    // PUT because it updates book repository
+    @RequestMapping(value = "/clear", method = RequestMethod.PUT)
+    public List<Book> clear() {
+        return bookRepository.dropAll();
+    }
+
+    @RequestMapping(value = "/title/{title}", method = RequestMethod.GET)
```

Comment:
If this API was specified in advance, that's fine, but anyway it is not consistent with REST methods you did above:

`GET /books` - lists all books
`GET /books/:id` - particular book with given ID
`DELETE /books/:id` - removes book
`PUT /books/:id` - alters book

Those are totally fine and common used as a REST. However, you can't do this (from REST definition perspective):

`GET /books/title/:title`

because of its filtering of existing collection and you suggest here that have some subresource called `title`. You probably should have something for filtering using query, like:

`GET /books?title=MyTitle`

And this approach is closer to REST definition.